### ISSUE TITLE
TypeError: unusable when reading request body in Node.js 20

### ISSUE THREAD

**Issue by lox** [mswjs/interceptors#419]
On Node.js v20.5.1 and the v0.25.1 of the interceptors, I get:

```
TypeError: unusable
    at Request.clone (node:internal/deps/undici/undici:7428:17)
```

When trying to clone a request and read the body:

```js
import { BatchInterceptor } from '@mswjs/interceptors'
import { ClientRequestInterceptor } from '@mswjs/interceptors/ClientRequest'
import { FetchInterceptor } from '@mswjs/interceptors/fetch'
import http from 'http'

const interceptor = new BatchInterceptor({
  name: 'my-interceptor',
  interceptors: [new ClientRequestInterceptor(), new FetchInterceptor()],
})

interceptor.apply()

interceptor.on('response', async ({ response, isMockedResponse, request, requestId }) => {
  console.log('response to %s %s was:', request.method, request.url, response.status)

  const respText = await response.clone().text()
  console.log('response', respText)

  const reqText = await request.clone().text()
  console.log('request', reqText)
})

const server = http.createServer((req, res) => {
  res.statusCode = 200
  res.setHeader('Content-Type', 'application/json')
  res.end(JSON.stringify({ method: req.method, headers: req.headers }))
})

server.listen(3000, '127.0.0.1', () => {
  console.log('Server running at http://127.0.0.1:3000/')
})

const data = JSON.stringify({
  hello: 'world',
})

const response = await fetch('http://localhost:3000/', {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
  },
  body: data,
})

const result = await response.json()
console.log('Result:', result)

interceptor.dispose()
server.close()
```

Am I missing something here? 

I am trying to write a record/playback layer, so I need to be able to read a copy of the request body without affecting the upstream request.

**Comment by kettanaito** [comment]
Hi, @lox. Thanks for reporting this.

The error itself originates from the `Request.clone()` method implementation in Undici:

https://github.com/nodejs/undici/blob/e39a6324c4474c6614cac98b8668e3d036aa6b18/lib/fetch/request.js#L715-L718

When attempting to `request.clone()`, that request's body has already been read. 

The `request` instance we provide in the `response` listener is the same request instance constructed and exposed to the `request` listener (called interactive request):

https://github.com/mswjs/interceptors/blob/45df4685cb31d2f66c37fcd60b0f53359d838c98/src/interceptors/ClientRequest/NodeClientRequest.ts#L250

Nothing should be ready this request body but the consumer. 

@lox, do you see this behavior happen in earlier versions of Node, like v18? I wonder if it's something specific to v20.

**Comment by lox** [comment]
I'll try earlier versions!

Any suggestions for alternative approaches?

**Comment by kettanaito** [comment]
Please let me know how it goes. We need to put this into an integration test if it proves to be reproducible on any supported Node.js version. We don't currently support Node.js 20 officially but I'd expect no significant changes to the Fetch API there. I certainly don't see any when looking at Undici. 

> Any suggestions for alternative approaches?

I think what you're doing is right with what Interceptors promises you. I'm considering if something along the way _does_ read the request body. I encountered a similar issue in the past and the solution there was to prepare multiple early clones of the same request and pass it to the relevant areas of the algorithm. I wouldn't go that route unless that proves to be the case. 

I wonder, can you try cloning the request in the `request` event instead? Would this make any difference? 

```js
const allRequests = new Map()

interceptor.on('request', async ({ request, requestId }) => {
  allRequests.set(requestId, request.clone())
})
interceptors.on('response', async ({ requestId }) => {
  const text = await allRequests.get(requestId).text()
})
```

**Comment by lox** [comment]
```
interceptor.on('request', async ({ request, requestId }) => {
  try {
    allRequests.set(requestId, request.clone())
  } catch (err) {
    console.log('err', err)
  }
})

interceptor.on('response', async ({ requestId }) => {
  try {
    const text = await allRequests.get(requestId).text()
    console.log('req', text)
  } catch (err) {
    console.log('err', err)
  }
})
```

This works!

**Comment by kettanaito** [comment]
Glad to hear that! And yet the code you were using initially should work too. I think there might be a bug in how we clone requests to expose them to the listeners.

**Comment by kettanaito** [comment]
I couldn't reproduce this on the latest version of the library. Please, if you still experience this, open a new issue and provide a link to a reproduction repo where I can inspect it. Thanks.

**Comment by kettanaito** [comment]
## Released: v0.34.3 🎉

This has been released in v0.34.3!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.34.3)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.34.3)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Comment by gsanta** [comment]
I still have this issue with interceptor v0.35.8
